Add vSphere Support Matrix to README

rguske opened the pull request:
Closes: #9

Add an overview about the supported vSphere versions to the main page. Therefore, our community can quickly see which Supervisor Services are supported for their vSphere environments.

![image](https://user-images.githubusercontent.com/31652019/225321498-1c5f4644-5a40-4e91-a5c7-9585df27aa32.png)

YuxinZhou reviewed (approved):
LGTM

YuxinZhou commented on README.md:
```diff
@@ -3,7 +3,19 @@
+| vSAN Data Persistence Platform Services - MinIO, Cloudian and Dell ObjectScale| ✅ | ✅ |
+| Backup \& Recovery Service - Velero | ✅ | ✅ |
+| Certificate Management Service - Cert-Manager | ✅ | ✅ |
+| Cloud Native Registry Service - Harbor | ❌ | ✅ |
```
On 7.0 (and 8.0 until the deprecation) we supported harbor as the embedded harbor registry. Should we add a footnote here? I feel the user may think there is no way to deploy harbor on 7.0 if they only look at the table.

rguske replied:
Valid point 👍🏻 We could put an * in the cell and underneath the table the footnote.

rguske replied:
@YuxinZhou @AcidLeroy @jswolf059 

How do you like:

<img width="1031" alt="image" src="https://user-images.githubusercontent.com/31652019/225728598-a4589767-55b0-4be7-ae0a-4adc6e5b1408.png">

fstrudel commented on README.md:
```diff
@@ -3,7 +3,19 @@
 New service will be added overtime with the goal to continue to empower your DevOps communities.
 
-*Supervisor Services are only available with Supervisor Clusters enabled using NSX-T*
+Supervisor Services **are only** available with Supervisor Clusters enabled **using NSX-T**.
```
Hmm that's not true though? PodVMs are supported on VDS (https://jira.eng.vmware.com/browse/VGL-1577) starting from 8.x (x=?) release, so we should beable to install services on VDS supervisors too.

rguske replied:
I haven't adjusted the original footnote since vSphere 8 Update 1 isn't GA'ed yet. Do you want the hint included already?
[Supervisor Services with vSphere Distributed Switch](https://core.vmware.com/resource/whats-new-vsphere-8-update-1#sec25426-sub4)

fstrudel replied:
@rguske Yes, since we announced it publicly already, I'd rather include it now?

fstrudel commented on README.md:
```diff
@@ -3,7 +3,19 @@
+| Certificate Management Service - Cert-Manager | ✅ | ✅ |
+| Cloud Native Registry Service - Harbor | ❌ | ✅ |
+| Kubernetes Ingress Controller Service - Contour | ❌ | ✅ |
+| Service Discovery Service - External DNS | ✅ | ✅ |
```
I'm surprised we call external DNS a "service discovery service"?

rguske replied:
I think this comes from the official descripion [HERE](https://github.com/kubernetes-sigs/external-dns).
> ExternalDNS makes Kubernetes resources discoverable via public DNS servers.

Alternatives?
- DNS Services
- DNS Automation Service
- DNS Lifecycle-Management Service

Just a couple of ideas.

fstrudel replied:
I'd say `Dynamic control of DNS records` instead.

rguske replied:
As a short description okay but as a Supervisor Service name? Like `Dynamic Control of DNS Records Service`?
Find it a little bit too long, isn't it?

fstrudel reviewed (approved):
I love the table! We might want to be more specific as I believe we've added more services to the "allow-list" between 8 and 8.x (like the cluster issuer).

rguske commented:
@fstrudel @YuxinZhou 
With my latest commit ac765ee I updated the support statement, added the hint for the embedded registry and adjusted the External DNS Service name.

douglas-c reviewed (commented):
The table looks great, thanks for putting this together!

Just a general FYI, the services implemented using Carvel packages are the ones that are not supported on vSphere 7.

rguske commented on README.md:
```diff
@@ -3,7 +3,19 @@
+| Certificate Management Service - Cert-Manager | ✅ | ✅ |
+| Cloud Native Registry Service - Harbor | ❌ * | ✅ |
+| Kubernetes Ingress Controller Service - Contour | ❌ | ✅ |
+| External DNS Service - External DNS | ✅ | ✅ |
```
Got it.

rguske commented on README.md:
```diff
@@ -3,7 +3,19 @@
+| :--: | :--: | :--: |
+| vSAN Data Persistence Platform Services - MinIO, Cloudian and Dell ObjectScale| ✅ | ✅ |
+| Backup \& Recovery Service - Velero | ✅ | ✅ |
+| Certificate Management Service - Cert-Manager | ✅ | ✅ |
```
If this is our support statement, sure. This means, that only DPP and Velero are supported for vSphere 7 and 8, right?

douglas-c replied:
Right, only the vSAN services and Velero work on vSphere 7.

rguske replied:
@douglas-c Adjustments made in 3619baf

rguske commented:
> Just a general FYI, the services implemented using Carvel packages are the ones that are not supported on vSphere 7.

I am not sure if I understand it fully correct. You are saying that only the Services which are added using the _Add Service_ feature are not supported with vSphere 7. So, the once which are available here on the catalog/repository.

douglas-c commented:
> I am not sure if I understand it fully correct. You are saying that only the Services which are added using the Add Service feature are not supported with vSphere 7. So, the once which are available here on the catalog/repository. 

No, that's not what I meant. The flow to register Services is the same no matter what. But the format of the YAML file which you pass to vCenter specifies which package manager it uses. Carvel packages include the "Package" kind in their manifest, which will not work on 7.x. Didn't mean to hold up the PR with this.

rguske commented:
> No, that's not what I meant. The flow to register Services is the same no matter what. But the format of the YAML file which you pass to vCenter specifies which package manager it uses. Carvel packages include the "Package" kind in their manifest, which will not work on 7.x. Didn't mean to hold up the PR with this.

I see. Thanks for the clarification. Please allow me an additional question. According to your answer, is it possible with vSphere 8 to register Services also in a different way than via using the provided `yaml` files of the Supervisor Catalog? That would be interesting and I'd like to know more about it :)
W

fstrudel commented:
> > No, that's not what I meant. The flow to register Services is the same no matter what. But the format of the YAML file which you pass to vCenter specifies which package manager it uses. Carvel packages include the "Package" kind in their manifest, which will not work on 7.x. Didn't mean to hold up the PR with this.
> 
> I see. Thanks for the clarification. Please allow me an additional question. According to your answer, is it possible with vSphere 8 to register Services also in a different way than via using the provided `yaml` files of the Supervisor Catalog? That would be interesting and I'd like to know more about it :) We can take this offline as well. Please go ahead with the PR if nothing else is missing.

Whether you're on 7 or 8, you need to provide a YAML that defines your package. In 7 we only support a YAML that contains a `SupervisorServiceDefinition`. In 8 we support a YAML that either contains `SupervisorServiceDefinition` or that contains `Package` and `PackageMetadata` (Carvel package).
So it depends want you mean by a "different way": it's one way in the sense that you need to upload a YAML file. It's multiple ways in the sense that we support 2 different types of content in this YAML file.
Does that help?

rguske commented:
> > > No, that's not what I meant. The flow to register Services is the same no matter what. But the format of the YAML file which you pass to vCenter specifies which package manager it uses. Carvel packages include the "Package" kind in their manifest, which will not work on 7.x. Didn't mean to hold up the PR with this.
> > 
> > 
> > I see. Thanks for the clarification. Please allow me an additional question. According to your answer, is it possible with vSphere 8 to register Services also in a different way than via using the provided `yaml` files of the Supervisor Catalog? That would be interesting and I'd like to know more about it :) We can take this offline as well. Please go ahead with the PR if nothing else is missing.
> 
> Whether you're on 7 or 8, you need to provide a YAML that defines your package. In 7 we only support a YAML that contains a `SupervisorServiceDefinition`. In 8 we support a YAML that either contains `SupervisorServiceDefinition` or that contains `Package` and `PackageMetadata` (Carvel package). So it depends want you mean by a "different way": it's one way in the sense that you need to upload a YAML file. It's multiple ways in the sense that we support 2 different types of content in this YAML file. Does that help?

Helps indeed @fstrudel , thy! I haven't seen any of provided information in our Docs, therefore I wasn't aware that this way of registering services exist as well.
